app/controllers/application_controller/timelines.rb

```diff
@@ -105,42 +105,54 @@ def tl_build_timeline_report_options
       end
 
       temp_clause = @tl_record.event_where_clause(@tl_options.evt_type)
+      conditions = [temp_clause[0], "timestamp >= ?", "timestamp <= ?"]
+      parameters = temp_clause[1..-1] + [from_dt, to_dt]
 
-      cond = "( "
-      cond = cond << temp_clause[0]
-      params = temp_clause.slice(1, temp_clause.length)
-
-      event_set = @tl_options.event_set
-      if !event_set.empty?
-        if @tl_options.policy_events? && @tl_options.policy.result != "both"
-          where_clause = [") and (timestamp >= ? and timestamp <= ?) and (event_type in (?)) and (result = ?)",
-                          from_dt,
-                          to_dt,
-                          event_set.flatten,
-                          @tl_options.policy.result]
-        else
-          where_clause = [") and (timestamp >= ? and timestamp <= ?) and (event_type in (?))",
-                          from_dt,
-                          to_dt,
-                          event_set.flatten]
-        end
-      else
-        where_clause = [") and (timestamp >= ? and timestamp <= ?)",
-                        from_dt,
-                        to_dt]
-      end
-      cond << where_clause[0]
+      tl_add_event_type_conditions(conditions, parameters)
```

Comment:
We could rename method to `tl_add_event_type_conditions!` to emphasise it modifies arguments inplace

Reply:
I did not feel the need to add `!` at the end of the method name, since there is no alternative method that would not modify its arguments. But I can add it if that would make things better.